pkg/compiler/check.go

```diff
@@ -204,7 +204,10 @@ func (comp *compiler) checkTypedefs() {
 			} else {
 				// For templates we only do basic checks of arguments.
 				names := make(map[string]bool)
-				for _, arg := range n.Args {
+				for i, arg := range n.Args {
+					if arg.Name == "BASE" && i != 0 {
```

Comment:
For all builtin types BASE is the last argument (or last but one if opt is present).

Reply:
Fixed.